src/instructlab/llamacpp/llamacpp_convert_to_gguf.py

```diff
@@ -1151,12 +1121,6 @@ def add_meta_arch(self, params: Params) -> None:
         self.gguf.add_head_count(params.n_head)
         self.gguf.add_head_count_kv(params.n_head_kv)
 
-        if params.n_experts:
```

Comment:
> which specific cases do you mean?

This code is deleted. And above params.n_experts was deleted.

Reply:
Let me double check. I *think* that this code was dead because of how params are loaded, but I will confirm.

Reply:
OK, the explanation as to why these two options are removed is here: 76410ae05cd81aeee305a48ed83da373d0ec28bf To quote,

```
    mypy: Remove use of undefined GGUFWriter symbols

    Neither add_expert_count nor add_expert_used_count are defined in the
    version of gguf library we use.  Actually, these symbols are not yet
    released on pypi, and are available in git repo of the library only:

    https://github.com/ggerganov/llama.cpp/commit/799a1cb13b0b1b560ab0ceff485caed68faa8f1f

    This makes mypy complain about it.

    This patch removes code related to these parameters. If and when we
    upgrade to gguf version that supports them, we may revert this patch.
```

Since GGUFWriter doesn't have the functions, passing these settings through config file would crash the code anyway. In the future, gguf may be updated to include these functions, then this may be revisited.

Let me know if this line of thought is ok for you.

Reply:
I guess I am worried were are changing 3rd party files just to make mypy happy. Perhaps we should exclude these 3rd party source files (e.g. this and the mlx_explore folder) from mypy checks.

Reply:
Depends on how you look at it. It's not just to make `mypy` happy: this patch removes a bug that results in a crash depending on what's in the config file. With this patch included, the unknown options will be silently ignored.

I was wondering about the nature of this file. The way these files are managed in this tree is ambiguous. On one hand, they clearly come from `llamacpp` repo. On the other hand, it's not pulled in through a library (which makes sense - since this code is NOT part of their package); and we don't track which particular version of the file we pulled here. (I actually tried to figure this out - and I suspect what we have here in-tree is a **modified** version of one of older versions of the `llamacpp` example file.)

We already fixed some bugs in this file, removed `click` code, and `blackened` / `ruffed` the file. (Just as examples. There are more.) If the decision is to not touch the file (and maybe some other files - which are these?), then we should document this (there's a README uder this path, we could update it.) Ideally, we would also document where this code comes from then (the exact commit message), or maybe even try to pull it through git submodules. (Not sure it's possible at this point though - the code diverged.)

(My personal opinion is that this would not be a good decision to make, because this code already diverged from wherever we took it from. But if the team feels otherwise, we could try to freeze new changes to the module, somehow, at least as a review policy.)

Reply:
@bjhargrave all patches for the modules you identified (plus one more) that are copy-pasted are dropped, series flattened. This should be less controversial, I think.